src/mountscript.sh

```diff
@@ -596,59 +639,429 @@ ensure_aznfswatchdog()
+                available_port=$(get_next_available_port)
+                if [ $? -ne 0 ]; then
+                        eecho "Failed to get the next available port for nfsv4 mount."
+                        return 1
+                fi
+                vecho "Next Available Port: '$available_port'"
+
+                if [ -z "$available_port" ]; then
+                        eecho "Running out of ports. Nfsv4 has port range $NFSV4_PORT_RANGE_START to $NFSV4_PORT_RANGE_END. All ports from this range are used by other processes."
+                        return 1
+                fi
+
+                used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+                vecho "used port: '$used_port'"
+
+                chattr -f -i $stunnel_conf_file
+
+                sed -i "s/$used_port/$available_port/" $stunnel_conf_file
+                if [ $? -ne 0 ]; then
+                        eecho "Failed to replace the port in $stunnel_conf_file."
+                        chattr -f +i $stunnel_conf_file
+                        return 1
+                fi
+                chattr -f +i $stunnel_conf_file
+
+                new_used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+
+                # start the stunnel process
+		stunnel_status=$(stunnel $stunnel_conf_file 2>&1)
+                if [ -n "$stunnel_status" ]; then
+                        # TODO: Change nfs_host to storageaccount later after server side changes are implemented and merged into Main.
+                        is_binding_error=$(echo $stunnel_status | grep "$LOCALHOST:$available_port: Address already in use")
+                        if [ -z "$is_binding_error" ]; then
+                                eecho "[FATAL] Not able to start stunnel process for '${stunnel_conf_file}'!"
+                                return 1
+                        fi
+		else
+			vecho "Found new port '$new_used_port' and restarted stunnel."
+			break
+                fi
+        done
+}
+
+get_next_available_port()
+{
+        for ((port=NFSV4_PORT_RANGE_START; port<=NFSV4_PORT_RANGE_END; port++))
+        do
+                is_port_available=`netstat -tulpn | grep ":$port "`
+                if [ -z "$is_port_available" ]; then
+                        break
+                fi
+        done
+
+        if [ $port -le $NFSV4_PORT_RANGE_END ]; then
+                echo "$port"
+        else
+                echo ""
+        fi
+}
+
+#
+# Add stunnel configuration in stunnel_<storageaccount>.conf file.
+#
+add_stunnel_configuration()
+{
+        chattr -f -i $stunnel_conf_file
+
+	stunnel_CAFile="$STUNNELDIR/certs/serverCert.pem"
+        echo "CAFile = $stunnel_CAFile" >> $stunnel_conf_file
+        if [ $? -ne 0 ]; then
+                chattr -f +i $stunnel_conf_file
+                eecho "Failed to add server certificate file path to $stunnel_conf_file!"
+                return 1
+        fi
+
+        echo "verify = 3" >> $stunnel_conf_file
```

Comment:
verify option can change depending on user personalization

Reply:
we are not using verify =3 option. We will be using verifyChain now